src/Microsoft.AspNetCore.Sockets/Transports/LongPollingTransport.cs

```diff
@@ -37,18 +37,19 @@ public LongPollingTransport(IReadableChannel<Message> connection, ILoggerFactory
                 // TODO: We need the ability to yield the connection without completing the channel.
                 // This is to force ReadAsync to yield without data to end to poll but not the entire connection.
                 // This is for cases when the client reconnects see issue #27
-                using (var message = await _connection.ReadAsync(context.RequestAborted))
+                await _application.WaitToReadAsync(context.RequestAborted);
```

Comment:
I was thinking about 204 too but the client MUST be able to handle 200 with no data since the server should timeout a pending poll if there was no messages for the corresponding connection within a timeout (we used 110 secs by default before because IIS would kill requests after 120 secs)